src/main/java/gwt/jsonix/marshallers/xjc/plugin/builders/MainJsBuilder.java

```diff
@@ -134,16 +139,20 @@ protected static void populateJCodeModel(JCodeModel toPopulate, JClass container
         addMarshall(toPopulate, mainJsClass, containerRef, callbackMap.get(MARSHALL_CALLBACK));
     }
 
-    protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate, String basePackage) throws JClassAlreadyExistsException {
-        String fullMainJsName = basePackage + "." + MAIN_JS;
+    protected static JDefinedClass getMainJsClass(final JCodeModel toPopulate, String basePackage, String mainJsName) throws JClassAlreadyExistsException {
+        String fullMainJsName = basePackage + "." + mainJsName;
         JDefinedClass jDefinedClass = toPopulate._getClass(fullMainJsName);
-
         if (jDefinedClass == null) {
-            jDefinedClass = toPopulate._class(basePackage + "." + MAIN_JS);
+            jDefinedClass = toPopulate._class(basePackage + "." + mainJsName);
```

Comment:
this can be rewritten as `jDefinedClass = toPopulate._class(fullMainJsName);`

Reply:
@jomarko 
Good catch, thx!